### PR TITLE
Adding list reservations API to Microsoft.Capacity

### PR DESCRIPTION
**Pull request by toddm**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Please ensure to add changelog with this PR by answering the following questions.
  1. What's the purpose of the update?    
      - [ ] new service onboarding 
      - [X] new API version 
      - [ ] update existing version for new feature 
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify 
  2. When you are targeting to deploy new service/feature to public regions? Please provide date, or month to public if date is not available yet. - By July 15, 2021
  3. When you expect to publish swagger? Please provide date, or month to public if date is not available yet. August 1, 2021
  4. If it's an update to existing version,  please select SDKs of specific language and CLIs that require refresh after swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [X] No, no need to refresh for updates in this PR

### Contribution checklist:
- [X] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [X] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [X] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [X] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  -  [ ] Ensure to copy the existing version into new directory structure for first commit (including refactoring) and then push new changes including version updates in separate commits. This is required to review the changes efficiently.
  - Adding a new service

- [X] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [X] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by chiragg4u** [comment]
Thank you for copying the existing version into new folder for first commit. It helps in reviewing the change.
I assume you followed the below guideline, please mark the checklist appropriately
Ensure to copy the existing version into new directory structure for first commit (including refactoring) and then push new changes including version updates in separate commits. This is required to review the changes efficiently.

**Comment by chiragg4u** [comment]
From the changes I don't see new API being added and lots of changes in terms of removing enum's etc. Please confirm if this is expected.

**Comment by chiragg4u** [comment]
I also see you've not copied all the paths in the new version, it will break your SDK's/CLI/PS. It's also highlighted in some of the automated checks, please closely review the automated checks and resolve.

**Comment by chiragg4u** [comment]
Changes looks good from ARM perspective, please follow the process given @ https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/212/Swagger-PR-Review to continue

**Comment by toddm** [comment]
@ArcturusZhang I have gotten all validations to pass except for go and python.  The errors they show don't make sense to me.  For example, it says there is a duplicate ReservationOrderList definition, but there is only one per api-version.
Are those fixable?  How do I proceed with this PR?

File: specification/reservations/resource-manager/readme.md
Comment (ArcturusZhang) [inline specification/reservations/resource-manager/readme.md:30]: We need to remove the same section at line 41-44, and the issues in go and python SDK generation should be gone.
We can only have one of these sections that specifies the default tag in `readme.md` for autorest

**Review by ArcturusZhang — CHANGES_REQUESTED** [review]

File: specification/reservations/resource-manager/readme.md
Comment (toddm) [reply specification/reservations/resource-manager/readme.md:30]: Thanks so much!

Comment (toddm) [reply specification/reservations/resource-manager/readme.md]: The go one is still failing with: FATAL: System.ArgumentException: An item with the same key has already been added. Key: Client

Comment (toddm) [reply specification/reservations/resource-manager/readme.md]: Strange.  By process of elimination, I figured out that it was the operation name of the new API.  I changed it and got it to pass locally.

**Comment by toddm** [comment]
@ArcturusZhang would you please review again

**Comment by toddm** [comment]
@jianyexi would you please review this?  There are breaking changes, but it's a new api-version so I think it's ok.

**Review by majastrz — APPROVED** [review]
Signed off from ARM side.

**Comment by jianyexi** [comment]
> @jianyexi Jianye Xi FTE would you please review this? There are breaking changes, but it's a new api-version so I think it's ok.



> @jianyexi Jianye Xi FTE would you please review this? There are breaking changes, but it's a new api-version so I think it's ok.

Why did you remove some paths , it's not allowed even in new api version ?

**Comment by toddm** [comment]
> > @jianyexi Jianye Xi FTE would you please review this? There are breaking changes, but it's a new api-version so I think it's ok.
> 
> Why did you remove some paths , it's not allowed even in new api version ?

I didn't remove any paths between stable versions.  There are some paths in a preview api-version that I didn't include in this new stable version.  Is that required?  I can bring them in if needed.

**Comment by toddm** [comment]
@jianyexi - would you take another look?  I added the paths that were in preview to this stable release.

**Comment by jianyexi** [comment]
> @jianyexi Jianye Xi FTE - would you take another look? I added the paths that were in preview to this stable release
Are these paths ready to merge into stable version ?

**Comment by toddm** [comment]
> > @jianyexi Jianye Xi FTE - would you take another look? I added the paths that were in preview to this stable release
> > Are these paths ready to merge into stable version ?

The paths are ready to merge into stable.  I put Do Not Merge in the title because we are still deploying the api-version and will then update the manifest, but it's otherwise ready.

File: specification/reservations/resource-manager/readme.md
Comment (ArcturusZhang) [inline specification/reservations/resource-manager/readme.md:38]: We did not include the `Microsoft.Capacity/stable/2020-10-25/quota.json` file - any reason here?

Comment (toddm) [reply specification/reservations/resource-manager/readme.md:38]: Just an oversight.  I've added it now.

**Review by ArcturusZhang — APPROVED** [review]

**Comment by toddm** [comment]
@jianyexi  We have made all of the necessary changes to support the new api-version, so we are now ready to merge this.

**Comment by jianyexi** [comment]
There are cross version breaking changes ,could you follow the action to get approval:
https://github.com/Azure/azure-rest-api-specs/pull/14849#issuecomment-902367089

**Comment by toddm** [comment]
> There are cross version breaking changes ,could you follow the action to get approval:
> [#14849 (comment)](https://github.com/Azure/azure-rest-api-specs/pull/14849#issuecomment-902367089)

@jianyexi, I reduced the cross-version breaking changes to only the return error message, which we're required to change for compliance.  Does that change the need to get a breaking change review?  The customer impact will be minimal.

**Comment by jianyexi** [comment]
> > There are cross version breaking changes ,could you follow the action to get approval:
> > [#14849 (comment)](https://github.com/Azure/azure-rest-api-specs/pull/14849#issuecomment-902367089)
> 
> @jianyexi Jianye Xi FTE, I reduced the cross-version breaking changes to only the return error message, which we're required to change for compliance. Does that change the need to get a breaking change review? The customer impact will be minimal.

The format change also need to be approved

**Comment by toddm** [comment]
@jianyexi can this be completed now?  I got the breaking change sign-off.

**Pull request merged**
